app/services/plugin.service.ts

```diff
@@ -2,139 +2,104 @@ import { Injectable, Inject } from '@angular/core';
-            case "Max Pooling":
-                return [["size", Type.Point]];
-            case "Reshape":
-                return [["shape", Type.String]];
-            case "Output":
-                return [];
-            default:
-                return [["beta", Type.Boolean]];
+    graphPluginData(kind: string) {
+        const type = this.definitions.graphs.get(kind);
+        if (!type) {
+            throw "type not found";
         }
+        return new PluginPropertyData(kind, type);
     }
-
-    edgePluginData(type: string) {
-        return new PluginPropertyData("Edge", []);
+    nodePluginData(kind: string) {
+        const type = this.definitions.nodes.get(kind);
+        if (!type) {
+            throw "type not found";
+        }
+        return new PluginPropertyData(kind, type);
+    }
+    edgePluginData(kind: string) {
+        const type = this.definitions.edges.get(kind);
+        if (!type) {
+            throw "type not found";
+        }
+        return new PluginPropertyData(kind, type);
     };
-
-    // getNodeComputedProperties(): Array<[string, Type.Type, (entity: PropertiedEntity) => void]> {
-    //     return [["Label", Type.String,
-    //         (th: PropertiedEntity) => {
-    //             let contentString = "";
-    //             switch (th.entityName) {
-    //                 case "Input":
-    //                     contentString = "Shape: " + th.pluginProperties.get("shape");
-    //                     break;
-    //                 case "Output":
-    //                 case "Fully Connected":
-    //                     break;
-    //                 case "Conv2D":
-    //                     contentString = "Stride: (" + th.pluginProperties.get("stride").x + ", " + th.pluginProperties.get("stride").y + ")\nOutput Depth: " + th.pluginProperties.get("output depth");
-    //                     break;
-    //                 case "Max Pooling":
-    //                     contentString = "Size: (" + th.pluginProperties.get("size").x + ", " + th.pluginProperties.get("size").y + ")";
-    //                     break;
-    //                 case "Reshape":
-    //                     contentString = "Shape: " + th.pluginProperties.get("shape");
-    //                     break;
-    //                 default:
-    //                     break;
-    //             }
-
-    //             return (th as any)["Label"] = th.entityName + "\n" + contentString;
-    //         }]];
-    // }
 }
 
 @Injectable()
 export class PluginService {
+    private plugins = new Map<string, Promise<ConcretePlugin>>();
     private interpretCode: Script;
     private runInputCode: Script;
-    private plugins: Map<string, ConcretePlugin> = new Map<string, ConcretePlugin>();
+    // TODO: load from somewhere
+    private pluginKinds = new Map([[MagicConstants.DFA_PLUGIN_KIND, { definitions: "./dfa-definition.sinapdef", interpreter: "./build/plugins/dfa-interpreter.js" }]])
```

Comment:
I don't think it will be necessary @RighteousRaichu since it will save the location of files to some sort of user data location.